sdk/Pulumi.Automation.Tests/LocalPulumiCommandTests.cs

```diff
@@ -0,0 +1,187 @@
+                await LocalPulumiCommand.Install(new LocalPulumiCommandOptions { Version = requestedVersion, Root = tempDir });
+                var home = Environment.GetFolderPath(Environment.SpecialFolder.UserProfile);
+                var pulumiBin = Path.Combine(home, ".pulumi", "versions", requestedVersion.ToString(), "bin", "pulumi");
+                FileInfo fi1 = new FileInfo(pulumiBin);
+                var t1 = fi1.CreationTime;
+                // Install again with the same options
+                await LocalPulumiCommand.Install(new LocalPulumiCommandOptions { Version = requestedVersion, Root = tempDir });
+                FileInfo fi2 = new FileInfo(pulumiBin);
+                var t2 = fi2.CreationTime;
+                Assert.Equal(t1, t2);
+            }
+            finally
+            {
+                Directory.Delete(tempDir, true);
+            }
+
+        }
+
+        [Fact]
+        public async Task VersionCheck()
+        {
+            var dirPath = Path.Combine(Path.GetTempPath(), "automation-test-" + Guid.NewGuid().ToString());
+            var dir = Directory.CreateDirectory(dirPath);
+            try
+            {
+                // Install an old version
+                var installed_version = new SemVersion(3, 99, 0);
+                await LocalPulumiCommand.Install(new LocalPulumiCommandOptions { Version = installed_version, Root = dirPath });
+
+                // Try to create a command with a more recent version
+                var requested_version = new SemVersion(3, 102, 0);
+                await Assert.ThrowsAsync<InvalidOperationException>(() => LocalPulumiCommand.CreateAsync(new LocalPulumiCommandOptions
+                {
+                    Version = requested_version,
+                    Root = dirPath
+                }));
+
+                // Opting out of the version check works
+                await LocalPulumiCommand.CreateAsync(new LocalPulumiCommandOptions
+                {
+                    Version = requested_version,
+                    Root = dirPath,
+                    SkipVersionCheck = true
+                });
+            }
+            finally
+            {
+                dir.Delete(true);
+            }
+        }
+
+        [Fact]
+        public void PulumiEnvironment()
+        {
+            var env = new Dictionary<string, string?>{
+                {"PATH", "/usr/bin"}
+            };
+            var newEnv = LocalPulumiCommand.PulumiEnvironment(env, "pulumi", false);
+            Assert.Equal("/usr/bin", newEnv["PATH"]);
+
+            env = new Dictionary<string, string?>{
+                {"PATH", "/usr/bin"}
+            };
+            newEnv = LocalPulumiCommand.PulumiEnvironment(env, "/some/install/root/bin/pulumi", false);
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                Assert.Equal("/some/install/root/bin;/usr/bin", newEnv["PATH"]);
+            }
```

Comment:
Do we have to be concerned about the file path separators on Windows?

Reply:
Yep! I initially thought only the `;` separator matters, but because we use `Path.GetDirectoryName(command)` we end up with a proper windows path, and this fails, which we can now see with `ci/test` https://github.com/pulumi/pulumi-dotnet/actions/runs/7698458332/job/20977797557.

I updated the test to make a clearer distinction https://github.com/pulumi/pulumi-dotnet/pull/226/commits/9b6a497ed2b20b3894f861c5c15c894e24a2a586